src/main/java/org/apache/sysds/runtime/controlprogram/federated/FederatedRange.java

```diff
@@ -81,6 +88,10 @@ public long getSize() {
 			size *= getSize(i);
 		return size;
 	}
+
+	public int getOverlapNum(){
+		return _overlapNum;
```

Comment:
In my experience it does not matter how many overlaps you have, just have a boolean specifying if there is overlap or not.

Reply:
This is to ensure that I can put similar ranges in a map without replacing any values. The range needs to be unique to achieve this.

Reply:
i don't get your argument, if you allocate different objects for each range, then there should not be any problems other than you have to change the compare method to also in case of same values look at object pointers.

Reply:
That would not be a good solution since it would affect all other federated ranges as well. The solution I implemented here will only affect the behavior of the overlapping ranges and will not change how the existing federated ranges behave. 
I did not really like this either when I added the _overlapNum, but it was a solution that would work for the case I added without negatively affecting anything else. 
Another option would be to generate a random ID instead of setting an overlap number, since I do not need this specific number anyway. The problem is that I also don't need a randomly generated ID, so it would be overkill to make the implementation when all I need is any integer. 
Another solution would be to add overlap as a boolean and then check if this boolean is true in compareTo, toString, equals, and hashCode. If it is true then I should include the object pointer in some way in the computation. But then the question is again if this is too much compared to just adding a unique integer from the beginning.

Reply:
I don't agree,

Actually now that i think about it, you would not need a boolean nor a overlapping index, since you are using the "PART" (soon to be "OVERLAP") to mark if a given federated matrix is overlapping, therefore you should not need any of this logic to handle any incoming logic, just switch on the federated type.

again the compare part is easily solved via object compare since it contains a "random" ID in form of its object pointer that is unique for each object.

Reply:
I would like to use the PART type as the condition for adding the object pointer to the aforementioned methods, but this is not available in FederatedRange. 

Are you sure about the name "OVERLAP"? Isn't it too long compared to the others? Additionally, it is not any kind of overlap. It is specific to the partial aggregates.

Reply:
well in compression the overlap state only happens on a right matrix multiplication, but in federated i can already imagine it happening many times, if you want to convince me on another name it is fine, but i think having the same name would be nice.

I guess you chose PART because it is a "PARTial aggregate value". I chose overlap because it is matrices that could be imagined to be on top of each other and added together to produce the result. I think that the name should reflect what it is, and if we have the same concept more places it should be called the same, that said i don't know what other ppl call it when they have distributed parts that could be reduced to the actual value.